### PR TITLE
Part2.1 to 2.10

### PR DESCRIPTION
**Pull request by prajitlal4**
- Created Course info app which displays list of Courses (2.1 - 2.5)
    - Can show any number of course and is split into 2 files.
 
- Created Phonebook app which displays list of contacts in a phonebook object. (2.6 - 2.10)
    - Able to add names and numbers if the same name doesnt exist in the phonebook already. 
    - Split into multiple components. Has filtering functionality.

### CODE REVIEW

File: part2/phonebook/src/App.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const [searchName, setSearchName] = useState('')
+  const [filter, setFilter] = useState(false)
+
+  const personToShow = filter
+    ? persons.filter(person => person.name.toLowerCase().includes(searchName.toLowerCase()))
+    : persons
+  
+  const addName = (event) => {
+    event.preventDefault()
+    console.log('button clicked', event.target)
```
Comment (ajmalhakeem) [inline part2/phonebook/src/App.js]: remove console.log

File: part2/phonebook/src/App.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    console.log(checkName)
```
Comment (ajmalhakeem) [inline part2/phonebook/src/App.js]: remove console.log

File: part2/phonebook/src/App.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      setNewNumber('')
+    }
+  }
+
+  const checkName = persons.some(person => {
+    if (person.name === newName) {
+      return true
+    }
+    return false
+  })
```
Comment (ajmalhakeem) [inline part2/phonebook/src/App.js]: declare variable before using it, javascript lets u do this way too but best practice is to declare a variable first

File: part2/phonebook/src/App.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  const handleNameChange = (event) => {
+    console.log(event.target.value)
```
Comment (ajmalhakeem) [inline part2/phonebook/src/App.js]: remove console.log

File: part2/phonebook/src/App.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    setNewName(event.target.value)
+  }
+  
+  const handleNumberChange = (event) => {
+    console.log(event.target.value)
```
Comment (ajmalhakeem) [inline part2/phonebook/src/App.js]: remove console.log

File: part2/phonebook/src/App.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    setNewNumber(event.target.value)
+  }
+
+  const handleSearchChange = (event) => {
+    console.log(event.target.value)
```
Comment (ajmalhakeem) [inline part2/phonebook/src/App.js]: remove console.log +1000000

File: part2/phonebook/src/components/PersonList.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+const PersonList = ({personList}) => {
+  const personToShow = personList
+
+  return (
+    <div>
+      {personToShow.map(person => 
+        <Person key={person.id} person={person} />
+      )}
+    </div>
```
Comment (ajmalhakeem) [inline part2/phonebook/src/components/PersonList.js]: why not just personList.map ?

File: part2/phonebook/src/components/Person.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+const Person = ({person}) => {
+  const name = person.name
+  const number = person.number
+
+  return (
+    <div>
+      <p>{name} {number}</p>
+    </div>
+  )
+}
```
Comment (ajmalhakeem) [inline part2/phonebook/src/components/Person.js]: an easier way to destructure this prop is `const {name, number} = person`

File: part2/courseinfo/src/components/Course.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
```
Comment (ajmalhakeem) [inline part2/courseinfo/src/components/Course.js:1]: u wanna split each of these components into their own files?

File: part2/courseinfo/src/components/Course.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+const Part = ({part}) => {
+  const name = part.name
+  const exercises = part.exercises
```
Comment (ajmalhakeem) [inline part2/courseinfo/src/components/Course.js]: ` const { name, exercises } = part`

File: part2/courseinfo/src/components/Course.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+const Header = ({header}) => {
+  return (
+    <h1>{header}</h1>
+  )
+}
+
+const Content = ({content}) => {
+  const parts = content
```
Comment (ajmalhakeem) [inline part2/courseinfo/src/components/Course.js]: add semi colons to avoid ambiguity

File: part2/courseinfo/src/components/Course.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+  return (
+    <p>
+    {parts.map(part=>
```
Comment (ajmalhakeem) [inline part2/courseinfo/src/components/Course.js]: `content.map...` ?

File: part2/courseinfo/src/components/Course.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+      <Part key={part.id} part={part}/>
+    )}
+  </p>
+  )
+}
+
+const Course = ({course}) => {
+  const total = course.parts.reduce((a, b) => {
+    return {exercises: a.exercises + b.exercises}
+  })
```
Comment (ajmalhakeem) [inline part2/courseinfo/src/components/Course.js]: what is this doing? just curious

File: part2/courseinfo/src/App.js [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+        }
+      ]
+    }
+  ]
+
+  return (
+    <div>
+    {courses.map(course=>
+      <Course key={course.id} course={course}/>
+    )}
```
Comment (ajmalhakeem) [inline part2/courseinfo/src/App.js]:
```
{courses.map((course)=>
      <Course key={course.id} course={course}/>
)}
```

File: part2/phonebook/src/components/PersonList.js (hunk 7 above)
Comment (prajitlal4) [reply part2/phonebook/src/components/PersonList.js]: forgot that the prop was the variable i was using, dont need to rename it to personToShow. will make change

File: part2/courseinfo/src/components/Course.js (hunk 9 above)
Comment (prajitlal4) [reply part2/courseinfo/src/components/Course.js:1]: I will work on this

File: part2/courseinfo/src/components/Course.js (hunk 13 above)
Comment (prajitlal4) [reply part2/courseinfo/src/components/Course.js]: im adding the total number of exercises for each part, the reduce function goes through each element

**Review by ajmalhakeem — APPROVED** [review]
lgtm

**Pull request merged**